golangci-lint installation fails

Running `make install-tools` fails with the following error:
```
GOBIN=/home/roeesh/the-mesh-for-data/./hack/tools/bin GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.19.1
go: finding github.com v1.19.1
go: finding github.com/golangci/golangci-lint/cmd/golangci-lint v1.19.1
go: finding github.com/golangci/golangci-lint/cmd v1.19.1
go: finding github.com/golangci v1.19.1
# github.com/golangci/golangci-lint/pkg/golinters
../go/pkg/mod/github.com/golangci/golangci-lint@v1.19.1/pkg/golinters/golint.go:50:14: l.LintASTFiles undefined (type *"github.com/golangci/lint-1".Linter has no field or method LintASTFiles)
make: *** [.mk/tools.mk:35: hack/tools/bin/golangci-lint] Error 2
```

Error is fixed when upgrading to golangci-lint v1.31.0

--- a/.mk/tools.mk
+++ b/.mk/tools.mk
@@ -32,7 +32,7 @@ $(TOOLBIN)/hugo:
 
 INSTALL_TOOLS += $(TOOLBIN)/golangci-lint
 $(TOOLBIN)/golangci-lint:
-	GOBIN=$(ABSTOOLBIN) GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.19.1
+	GOBIN=$(ABSTOOLBIN) GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.31.0
 	$(call post-install-check)
 
 INSTALL_TOOLS += $(TOOLBIN)/kubebuilder

--- a/manager/controllers/app/blueprint_controller.go
+++ b/manager/controllers/app/blueprint_controller.go
@@ -181,7 +181,7 @@ func (r *BlueprintReconciler) reconcile(ctx context.Context, log logr.Logger, bl
 		}
 
 		numReleases++
-		//check the release status
+		// check the release status
 		rel, err := r.Helmer.Status(blueprint.Namespace, releaseName)
 		// unexisting release or a failed release - re-apply the chart
 		if updateRequired || err != nil || rel == nil || rel.Info.Status == release.StatusFailed {

--- a/manager/controllers/app/blueprint_controller_test.go
+++ b/manager/controllers/app/blueprint_controller_test.go
@@ -14,7 +14,6 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
-	//"github.com/ibm/the-mesh-for-data/manager/controllers/utils"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/yaml"

--- a/manager/controllers/app/m4dapplication_controller.go
+++ b/manager/controllers/app/m4dapplication_controller.go
@@ -169,7 +169,7 @@ func (r *M4DApplicationReconciler) deleteExternalResources(applicationContext *a
 	if err := r.DeleteOwnedBlueprint(applicationContext); err != nil {
 		return err
 	}
-	//delete the allocated namespace
+	// delete the allocated namespace
 	if err := r.DeleteNamespace(namespace); err != nil {
 		return err
 	}

--- a/manager/controllers/app/moduleinstance.go
+++ b/manager/controllers/app/moduleinstance.go
@@ -122,7 +122,7 @@ func (r *M4DApplicationReconciler) SelectModuleInstancesPerDataset(item modules.
 		return instances
 	}
 
-	//If sources of this module include source, copy is not required
+	// If sources of this module include source, copy is not required
 	r.Log.V(0).Info("Checking supported read sources")
 	sources := GetSupportedReadSources(readSelector.GetModule())
 	utils.PrintStructure(sources, r.Log, "Read sources")

--- a/manager/controllers/app/modules/moduleinterface.go
+++ b/manager/controllers/app/modules/moduleinterface.go
@@ -18,7 +18,7 @@ type Transformations struct {
 	Reason             string
 }
 
-//DataInfo defines all the information about the given data set
+// DataInfo defines all the information about the given data set
 type DataInfo struct {
 	// Data asset unique identifier, not necessarily the same string appearing in the resource definition
 	AssetID string
@@ -84,7 +84,7 @@ func (m *Selector) AddModuleInstances(args *app.ModuleArguments, item DataInfo) 
 	return instances
 }
 
-//SelectModule finds the module that fits the requirements
+// SelectModule finds the module that fits the requirements
 func (m *Selector) SelectModule(moduleMap map[string]*app.M4DModule) bool {
 	m.Message = ""
 	for _, module := range moduleMap {

--- a/manager/controllers/app/storage.go
+++ b/manager/controllers/app/storage.go
@@ -17,7 +17,7 @@ import (
 // +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dbuckets,verbs=get;list;watch;update;
 // +kubebuilder:rbac:groups=app.m4d.ibm.com,resources=m4dbuckets/status,verbs=get;update;patch
 
-//FindAvailableBucket finds an available storage asset
+// FindAvailableBucket finds an available storage asset
 func (r *M4DApplicationReconciler) FindAvailableBucket(owner types.NamespacedName, id string, prefixBase string, canShare bool) (*app.M4DBucket, error) {
 	ctx := context.Background()
 

--- a/manager/controllers/motion/suite_test.go
+++ b/manager/controllers/motion/suite_test.go
@@ -107,9 +107,9 @@ var _ = BeforeSuite(func(done Done) {
 
 	Expect(k8sClient).ToNot(BeNil())
 
-	//k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
-	//Expect(err).ToNot(HaveOccurred())
-	//Expect(k8sClient).ToNot(BeNil())
+	// k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	// Expect(err).ToNot(HaveOccurred())
+	// Expect(k8sClient).ToNot(BeNil())
 
 	close(done)
 }, 60)

--- a/manager/controllers/utils/utils.go
+++ b/manager/controllers/utils/utils.go
@@ -33,7 +33,7 @@ func DetermineCause(err error, prefix string) string {
 	}
 }
 
-//GetDataFormat returns the existing data format
+// GetDataFormat returns the existing data format
 func GetDataFormat(info *dc.DatasetDetails) (app.DataFormatType, error) {
 	switch info.DataFormat {
 	case "parquet":
@@ -54,7 +54,7 @@ func GetDataFormat(info *dc.DatasetDetails) (app.DataFormatType, error) {
 	return app.Binary, errors.New("Unknown format " + info.DataFormat)
 }
 
-//GetProtocol returns the existing data protocol
+// GetProtocol returns the existing data protocol
 func GetProtocol(info *dc.DatasetDetails) (app.IFProtocol, error) {
 	switch info.DataStore.Type {
 	case dc.DataStore_S3:
@@ -69,17 +69,17 @@ func GetProtocol(info *dc.DatasetDetails) (app.IFProtocol, error) {
 
 // IsTransformation returns true if the data transformation is required
 func IsTransformation(actionName string) bool {
-	return (actionName != "Allow") //TODO FIX THIS
+	return (actionName != "Allow") // TODO FIX THIS
 }
 
 // IsAction returns true if any action is required
 func IsAction(actionName string) bool {
-	return (actionName != "Allow") //TODO FIX THIS
+	return (actionName != "Allow") // TODO FIX THIS
 }
 
 // IsDenied returns true if the data access is denied
 func IsDenied(actionName string) bool {
-	return (actionName == "Deny") //TODO FIX THIS
+	return (actionName == "Deny") // TODO FIX THIS
 }
 
 // GetAttribute parses a JSON string and returns the required attribute value

--- a/manager/main.go
+++ b/manager/main.go
@@ -97,13 +97,12 @@ func main() {
 		os.Exit(1)
 	}
 
-	//Initialize PolicyCompiler interface
+	// Initialize PolicyCompiler interface
 	policyCompiler := pc.NewPolicyCompiler()
 
 	// Initiate the M4DApplication Controller (aka Pilot)
 	applicationController := app.NewM4DApplicationReconciler(mgr, "M4DApplication", vaultClient, policyCompiler)
 	if err := applicationController.SetupWithManager(mgr); err != nil {
-		//if err := controllers.NewM4DApplicationController(mgr, "M4DApplication"); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "M4DApplication")
 		os.Exit(1)
 	}

--- a/pkg/policy-compiler/pc-bl/combined_policies_decisions.go
+++ b/pkg/policy-compiler/pc-bl/combined_policies_decisions.go
@@ -10,15 +10,16 @@ import (
 )
 
 func GetCombinedPoliciesDecisions(firstDecisions *pb.PoliciesDecisions, secondDecisions *pb.PoliciesDecisions) *pb.PoliciesDecisions {
-	//Create a map of maps
+	// Create a map of maps
 	combinedDecisions := make(map[string]map[[2]string]*pb.OperationDecision)
 
-	//Populate using firstDecisions
+	// Populate using firstDecisions
 	PopulateMapWithDecisions(firstDecisions, combinedDecisions)
-	//Now populate with secondDecisions
+
+	// Now populate with secondDecisions
 	PopulateMapWithDecisions(secondDecisions, combinedDecisions)
 
-	//Now create the object to return
+	// Now create the object to return
 	toReturn := &pb.PoliciesDecisions{DatasetDecisions: []*pb.DatasetDecision{}, ComponentVersions: append(firstDecisions.GetComponentVersions(), secondDecisions.GetComponentVersions()...), GeneralDecisions: append(firstDecisions.GetGeneralDecisions(), secondDecisions.GetGeneralDecisions()...)}
 
 	for datasetIdentifier := range combinedDecisions {
@@ -39,16 +40,16 @@ func PopulateMapWithDecisions(decisions *pb.PoliciesDecisions, combinedDecisions
 		currDatasetIdentifier := datasetDecision.GetDataset().GetDatasetId()
 
 		if _, ok := combinedDecisions[currDatasetIdentifier]; !ok {
-			//Add the key
+			// Add the key
 			combinedDecisions[currDatasetIdentifier] = make(map[[2]string]*pb.OperationDecision)
 		}
 
-		//Iterate over operationDecisions
+		// Iterate over operationDecisions
 		for _, operationDecision := range datasetDecision.GetDecisions() {
 			currAccessOperation := [2]string{strconv.Itoa(int(operationDecision.GetOperation().GetType())), operationDecision.GetOperation().GetDestination()}
 
 			if _, ok := combinedDecisions[currDatasetIdentifier][currAccessOperation]; !ok {
-				//Add the key
+				// Add the key
 				combinedDecisions[currDatasetIdentifier][currAccessOperation] = &pb.OperationDecision{Operation: operationDecision.GetOperation(), EnforcementActions: operationDecision.GetEnforcementActions(), UsedPolicies: operationDecision.GetUsedPolicies()}
 			} else {
 				tempOperationDecision := combinedDecisions[currDatasetIdentifier][currAccessOperation]

--- a/pkg/policy-compiler/pc-bl/combined_policies_decisions_test.go
+++ b/pkg/policy-compiler/pc-bl/combined_policies_decisions_test.go
@@ -28,12 +28,12 @@ func checkCombinedPolicies(t *testing.T, decisions1, decisions2 *pb.PoliciesDeci
 	for _, datasetDecisions := range decisions2.DatasetDecisions {
 		tu.VerifyContainsDatasetDecision(t, combinedPolicies, datasetDecisions)
 	}
-	//here could be verification on order if we would work with it
-	//todo: check duplication of same policy/action
+	// here could be verification on order if we would work with it
+	// TODO: check duplication of same policy/action
 }
 
-//TestSingleColumnSingleOperationCombine checks if multiple enforcement actions combined
-//together performed on a single column is returned correctly in accordance with what is expected
+// TestSingleColumnSingleOperationCombine checks if multiple enforcement actions combined
+// together performed on a single column is returned correctly in accordance with what is expected
 
 func TestSingleColumnSingleOperationCombine(t *testing.T) {
 	operationDecision1 := &pb.OperationDecision{Operation: &pb.AccessOperation{Type: pb.AccessOperation_READ},
@@ -52,9 +52,9 @@ func TestSingleColumnSingleOperationCombine(t *testing.T) {
 		&pb.PoliciesDecisions{DatasetDecisions: []*pb.DatasetDecision{datasetDecison2}})
 }
 
-//TestSingleDatasetSingleOperationCombine checks if multiple enforcement actions combined
-//together and performed on multiple columns pertaining to one single dataset
-//then if the policies are returned correctly in accordance with what is expected
+// TestSingleDatasetSingleOperationCombine checks if multiple enforcement actions combined
+// together and performed on multiple columns pertaining to one single dataset
+// then if the policies are returned correctly in accordance with what is expected
 
 func TestSingleDatasetSingleOperationCombine(t *testing.T) {
 	operationDecision1 := &pb.OperationDecision{Operation: &pb.AccessOperation{Type: pb.AccessOperation_READ},
@@ -73,9 +73,9 @@ func TestSingleDatasetSingleOperationCombine(t *testing.T) {
 		&pb.PoliciesDecisions{DatasetDecisions: []*pb.DatasetDecision{datasetDecison2}})
 }
 
-//TestSingleColumnTwoOperationsCombine checks if two operations are combined
-//together and performed on single columns  then if the copmbined  policies are returned
-//correctly in accordance with what is expected
+// TestSingleColumnTwoOperationsCombine checks if two operations are combined
+// together and performed on single columns  then if the copmbined  policies are returned
+// correctly in accordance with what is expected
 
 func TestSingleColumnTwoOperationsCombine(t *testing.T) {
 	operationDecision1 := &pb.OperationDecision{Operation: &pb.AccessOperation{Type: pb.AccessOperation_COPY},
@@ -94,9 +94,9 @@ func TestSingleColumnTwoOperationsCombine(t *testing.T) {
 		&pb.PoliciesDecisions{DatasetDecisions: []*pb.DatasetDecision{datasetDecison2}})
 }
 
-//TestTwoDatasetsSingleOperationCombine checks if single operations is performed on multiple
-//datasets and then the policies are combined together then if the combined policies are
-//returned correctly in accordance with what is expected
+// TestTwoDatasetsSingleOperationCombine checks if single operations is performed on multiple
+// datasets and then the policies are combined together then if the combined policies are
+// returned correctly in accordance with what is expected
 
 func TestTwoDatasetsSingleOperationCombine(t *testing.T) {
 	operationDecision1 := &pb.OperationDecision{Operation: &pb.AccessOperation{Type: pb.AccessOperation_COPY},

--- a/pkg/policy-compiler/pc-bl/policy_compiler.go
+++ b/pkg/policy-compiler/pc-bl/policy_compiler.go
@@ -11,7 +11,7 @@ import (
 	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
 )
 
-//PolicyCompiler iassumed to have main policies-manager connector and may have or not extension policies-manager
+// PolicyCompiler iassumed to have main policies-manager connector and may have or not extension policies-manager
 type PolicyCompiler struct {
 	mainPolicyManagerCon       PolicyManagerConnector
 	extensionsPolicyManagerCon PolicyManagerConnector

--- a/pkg/policy-compiler/pc-bl/policy_compiler_test.go
+++ b/pkg/policy-compiler/pc-bl/policy_compiler_test.go
@@ -57,13 +57,13 @@ func policyCompilerMainAndExtReversed(purpose string) *pb.PoliciesDecisions {
 	return policies
 }
 
-//Tests  GetEnforcementActions in policy_compiler.go
-//tests the main policy manager configuration
-//test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
+// Tests  GetEnforcementActions in policy_compiler.go
+// tests the main policy manager configuration
+// test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
 func TestMainPolicyCompiler(t *testing.T) {
 	policyDecision := policyCompilerMainOnly("fraud-detection")
 	mainPMpolicies := tu.GetMainPMDecisions("fraud-detection")
-	//we add component version similar to the one added by PolicyCompiler
+	// we add component version similar to the one added by PolicyCompiler
 	serverComponentVersion := &pb.ComponentVersion{Name: "PCComponent", Id: "ID-1", Version: "1.0"}
 	mainPMpolicies.ComponentVersions = append(mainPMpolicies.ComponentVersions, serverComponentVersion)
 	tu.EnsureDeepEqualDecisions(t, policyDecision, mainPMpolicies)
@@ -74,13 +74,13 @@ func TestMainPolicyCompiler(t *testing.T) {
 	tu.EnsureDeepEqualDecisions(t, policyDecision2, mainPMpolicies2)
 }
 
-//Tests  GetEnforcementActions in policy_compiler.go
-//tests the extension policy manager configuration (used instead on main policy-maanager)
-//test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
+// Tests  GetEnforcementActions in policy_compiler.go
+// tests the extension policy manager configuration (used instead on main policy-maanager)
+// test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
 func TestExtPolicyCompiler(t *testing.T) {
 	policyDecision := policyCompilerExtOnly("fraud-detection")
 	extPMpolicies := tu.GetExtPMDecisions("fraud-detection")
-	//we add component version similar to the one added by PolicyCompiler
+	// we add component version similar to the one added by PolicyCompiler
 	serverComponentVersion := &pb.ComponentVersion{Name: "PCComponent", Id: "ID-1", Version: "1.0"}
 	extPMpolicies.ComponentVersions = append(extPMpolicies.ComponentVersions, serverComponentVersion)
 	tu.EnsureDeepEqualDecisions(t, policyDecision, extPMpolicies)
@@ -91,9 +91,9 @@ func TestExtPolicyCompiler(t *testing.T) {
 	tu.EnsureDeepEqualDecisions(t, policyDecision2, extPMpolicies2)
 }
 
-//Tests  GetEnforcementActions in policy_compiler.go
-//tests main policy manager configuration combined with extension policy manager
-//test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
+// Tests  GetEnforcementActions in policy_compiler.go
+// tests main policy manager configuration combined with extension policy manager
+// test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
 func TestMainAndExtPolicyCompiler(t *testing.T) {
 	policyDecision := policyCompilerMainAndExt("fraud-detection")
 	mainPMpolicies := tu.GetMainPMDecisions("fraud-detection")
@@ -106,9 +106,9 @@ func TestMainAndExtPolicyCompiler(t *testing.T) {
 	tu.CheckPolicies(t, policyDecision2, mainPMpolicies2, extPMpolicies2)
 }
 
-//Tests  GetEnforcementActions in policy_compiler.go
-//tests main policy manager configuration combined with extension policy manager - after we switch the main and the extension one
-//test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
+// Tests  GetEnforcementActions in policy_compiler.go
+// tests main policy manager configuration combined with extension policy manager - after we switch the main and the extension one
+// test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
 func TestMainAndExtPolicyCompilerReversed(t *testing.T) {
 	policyDecision := policyCompilerMainAndExtReversed("fraud-detection")
 	mainPMpolicies := tu.GetMainPMDecisions("fraud-detection")
@@ -121,7 +121,7 @@ func TestMainAndExtPolicyCompilerReversed(t *testing.T) {
 	tu.CheckPolicies(t, policyDecision2, mainPMpolicies2, extPMpolicies2)
 }
 
-//TestMain executes the above defined test function.
+// TestMain executes the above defined test function.
 
 func TestMain(m *testing.M) {
 	fmt.Println("TestMain function called = policy_compiler_test ")

--- a/pkg/policy-compiler/pc-bl/policy_manager_test.go
+++ b/pkg/policy-compiler/pc-bl/policy_manager_test.go
@@ -28,36 +28,36 @@ func policyManagerExtOnly(purpose string) *pb.PoliciesDecisions {
 	return policiesDecision
 }
 
-//Tests GetEnforcementActions func in policy_manager.go
-//tests the main policy manager configuration
-//test for purpose "fraud-detection", connector mocks configured for specific output per purpose
+// Tests GetEnforcementActions func in policy_manager.go
+// tests the main policy manager configuration
+// test for purpose "fraud-detection", connector mocks configured for specific output per purpose
 func TestMainPolicyManagerPurpose1(t *testing.T) {
 	policyDecision := policyManagerMainOnly("fraud-detection")
 	mainPMpolicies := tu.GetMainPMDecisions("fraud-detection")
 	tu.EnsureDeepEqualDecisions(t, policyDecision, mainPMpolicies)
 }
 
-//Tests GetEnforcementActions func in policy_manager.go
-//tests the main policy manager configuration
-//test for purpose "marketing", connector mocks configured for specific output per purpose
+// Tests GetEnforcementActions func in policy_manager.go
+// tests the main policy manager configuration
+// test for purpose "marketing", connector mocks configured for specific output per purpose
 func TestMainPolicyManagerPurpose2(t *testing.T) {
 	policyDecision := policyManagerMainOnly("marketing")
 	mainPMpolicies := tu.GetMainPMDecisions("marketing")
 	tu.EnsureDeepEqualDecisions(t, policyDecision, mainPMpolicies)
 }
 
-//Tests GetEnforcementActions func in policy_manager.go
-//tests the extensions policy manager configuration
-//test for purpose "fraud-detection", connector mocks configured for specific output per purpose
+// Tests GetEnforcementActions func in policy_manager.go
+// tests the extensions policy manager configuration
+// test for purpose "fraud-detection", connector mocks configured for specific output per purpose
 func TestExtPolicyManagerPurpose1(t *testing.T) {
 	policyDecision := policyManagerExtOnly("fraud-detection")
 	extPMpolicies := tu.GetExtPMDecisions("fraud-detection")
 	tu.EnsureDeepEqualDecisions(t, policyDecision, extPMpolicies)
 }
 
-//Tests GetEnforcementActions func in policy_manager.go
-//tests the extensions policy manager configuration
-//test for purpose "marketing", connector mocks configured for specific output per purpose
+// Tests GetEnforcementActions func in policy_manager.go
+// tests the extensions policy manager configuration
+// test for purpose "marketing", connector mocks configured for specific output per purpose
 func TestExtPolicyManagerPurpose2(t *testing.T) {
 	policyDecision := policyManagerExtOnly("marketing")
 	extPMpolicies := tu.GetExtPMDecisions("marketing")

--- a/pkg/policy-compiler/policy-compiler/policy_compiler_server_test.go
+++ b/pkg/policy-compiler/policy-compiler/policy_compiler_server_test.go
@@ -30,9 +30,9 @@ func verifyWrapperPolicyCompiler(t *testing.T, policyCompiler *bl.PolicyCompiler
 	tu.EnsureDeepEqualDecisions(t, decisions, innerDecisions)
 }
 
-//Tests  GetPoliciesDecisions in policy_compiler_server.go
-//tests the extension policy manager configuration (used instead on main policy-maanager)
-//test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
+// Tests  GetPoliciesDecisions in policy_compiler_server.go
+// tests the extension policy manager configuration (used instead on main policy-maanager)
+// test for purpose "fraud-detection" and "marketing" purposes, connector mocks configured for different outputs for these purposes
 func TestExtPolicyCompiler(t *testing.T) {
 	mainPolicyManager, extPolicyManager := constructPolicyConnectors()
 

--- a/pkg/policy-compiler/testutil/helper.go
+++ b/pkg/policy-compiler/testutil/helper.go
@@ -6,11 +6,12 @@ package testutil
 import (
 	"context"
 	"fmt"
-	"github.com/ibm/the-mesh-for-data/manager/controllers/utils"
 	"log"
 	"net"
 	"strconv"
 	"testing"
+
+	"github.com/ibm/the-mesh-for-data/manager/controllers/utils"
 
 	"github.com/golang/protobuf/proto"
 	"github.com/stretchr/testify/assert"
@@ -97,7 +98,7 @@ func VerifyContainsDatasetDecision(t *testing.T, combinedPolicies *pb.PoliciesDe
 			for _, operationDecision := range datasetDecison.Decisions {
 				VerifyContainsSingleOperationDesision(t, combinedDatasetPolicies, operationDecision)
 			}
-			//found correct dataset and all operation-desioins inside it
+			// found correct dataset and all operation-desioins inside it
 			return
 		}
 	}
@@ -107,7 +108,7 @@ func VerifyContainsDatasetDecision(t *testing.T, combinedPolicies *pb.PoliciesDe
 func VerifyContainsSingleOperationDesision(t *testing.T, combinedDatasetPolicies *pb.DatasetDecision, operationDesicion *pb.OperationDecision) {
 	for _, decision := range combinedDatasetPolicies.Decisions {
 		if proto.Equal(decision.Operation, operationDesicion.Operation) {
-			//found correct decision for this dataset and access operation
+			// found correct decision for this dataset and access operation
 			for _, action := range operationDesicion.EnforcementActions {
 				isFound := false
 				for _, actionCombined := range decision.EnforcementActions {
@@ -117,7 +118,7 @@ func VerifyContainsSingleOperationDesision(t *testing.T, combinedDatasetPolicies
 					}
 				}
 				if !isFound {
-					//one of the actions in dataset action is not in combined decisions
+					// one of the actions in dataset action is not in combined decisions
 					assert.Fail(t, "combined desisions miss enforcement action ", action)
 				}
 			}
@@ -130,11 +131,11 @@ func VerifyContainsSingleOperationDesision(t *testing.T, combinedDatasetPolicies
 					}
 				}
 				if !isFound {
-					//one of the usedPolicies in dataset action is not in combined decisions
+					// one of the usedPolicies in dataset action is not in combined decisions
 					assert.Fail(t, "combined desisions miss used policy ", usedPolicy)
 				}
 			}
-			return //found correct desision and it contains all enforceemnt actions and used policies
+			return // found correct desision and it contains all enforceemnt actions and used policies
 		}
 	}
 	assert.Fail(t, "didn't find the correct deision for this operation ", operationDesicion.Operation)
@@ -150,7 +151,7 @@ func CheckPolicies(t *testing.T, policies *pb.PoliciesDecisions, decision1, deci
 }
 
 /****************************/
-//Connector Mock "Main policy manager"
+// Connector Mock "Main policy manager"
 
 type connectorMockMain struct {
 	pb.UnimplementedPolicyManagerServiceServer
@@ -200,7 +201,7 @@ func MockMainConnector(port int) {
 }
 
 /****************************/
-//Connector Mock "Extension policy manager"
+// Connector Mock "Extension policy manager"
 
 type connectorMockExt struct {
 	pb.UnimplementedPolicyManagerServiceServer

--- a/samples/gui/server/datauser/utils.go
+++ b/samples/gui/server/datauser/utils.go
@@ -73,4 +73,4 @@ func ErrRender(err error) render.Renderer {
 	}
 }
 
-//var ErrNotFound = &ErrResponse{HTTPStatusCode: 404, StatusText: "Resource not found."}
+// var ErrNotFound = &ErrResponse{HTTPStatusCode: 404, StatusText: "Resource not found."}

--- a/test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go
+++ b/test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go
@@ -14,6 +14,7 @@ import (
 
 	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
 
+	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/status"
 )
@@ -26,7 +27,7 @@ func getEnv(key string) string {
 	return value
 }
 
-func GetMetadata(datasetID string) {
+func GetMetadata(datasetID string) error {
 	catalogConnectorURL := getEnv("CATALOG_CONNECTOR_URL")
 	catalogProviderName := getEnv("CATALOG_PROVIDER_NAME")
 
@@ -34,7 +35,7 @@ func GetMetadata(datasetID string) {
 	timeoutInSeconds, err := strconv.Atoi(timeoutInSecs)
 	if err != nil {
 		log.Printf("Atoi conversion of timeoutinseconds failed: %v", err)
-		return
+		return errors.Wrap(err, "Atoi conversion of timeoutinseconds failed")
 	}
 
 	fmt.Println("timeoutInSeconds: ", timeoutInSeconds)
@@ -49,7 +50,7 @@ func GetMetadata(datasetID string) {
 		errStatus, _ := status.FromError(err)
 		fmt.Println(errStatus.Message())
 		fmt.Println(errStatus.Code())
-		return
+		return errors.Wrap(err, "Connection to "+catalogProviderName+" Catalog Connector failed")
 	}
 	defer conn.Close()
 
@@ -74,15 +75,7 @@ func GetMetadata(datasetID string) {
 		fmt.Println("Message:", errStatus.Message())
 		// lets print the error code which is `INVALID_ARGUMENT`
 		fmt.Println("Code:", errStatus.Code())
-		// Want its int version for some reason?
-		// you shouldn't actullay do this, but if you need for debugging,
-		// you can do `int(status_code)` which will give you `3`
-		//
-		// Want to take specific action based on specific error?
-		// if codes.InvalidArgument == errStatus.Code() {
-		// 	do your stuff here
-		// 	log.Fatal()
-		// }
+		return errors.Wrap(err, "Error sending data to Catalog Connector")
 	} else {
 		fmt.Println("***************************************************************")
 		log.Printf("Received Response for GetDatasetInfo with  datasetID: %s\n", r.GetDatasetId())
@@ -91,10 +84,11 @@ func GetMetadata(datasetID string) {
 		s, _ := json.MarshalIndent(r, "", "\t")
 		fmt.Print(string(s))
 		fmt.Println("***************************************************************")
+		return nil
 	}
 }
 
-func GetCredentials(datasetID string) {
+func GetCredentials(datasetID string) error {
 	credentialsConnectorURL := getEnv("CREDENTIALS_CONNECTOR_URL")
 	credentialsProviderName := getEnv("CREDENTIALS_PROVIDER_NAME")
 
@@ -103,7 +97,7 @@ func GetCredentials(datasetID string) {
 
 	if err != nil {
 		log.Printf("Atoi conversion of timeoutinseconds failed: %v", err)
-		return
+		return errors.Wrap(err, "Atoi conversion of timeoutinseconds failed in GetCredentials")
 	}
 
 	fmt.Println("timeoutInSeconds: ", timeoutInSeconds)
@@ -119,7 +113,7 @@ func GetCredentials(datasetID string) {
 		fmt.Println(errStatus.Message())
 
 		fmt.Println(errStatus.Code())
-		return
+		return errors.Wrap(err, "Connection to Credentials Connector failed")
 	}
 	defer conn.Close()
 
@@ -144,15 +138,7 @@ func GetCredentials(datasetID string) {
 		log.Println("Message:", errCredentialStatus.Message())
 		// lets print the error code which is `INVALID_ARGUMENT`
 		log.Println("Code:", errCredentialStatus.Code())
-		// Want its int version for some reason?
-		// you shouldn't actullay do this, but if you need for debugging,
-		// you can do `int(status_code)` which will give you `3`
-		//
-		// Want to take specific action based on specific error?
-		// if codes.InvalidArgument == errCredentialStatus.Code() {
-		// 	do your stuff here
-		// 	log.Fatal()
-		// }
+		return errors.Wrap(err, "Error sending data to Credentials Connector in GetCredentials")
 	} else {
 		log.Println("***************************************************************")
 		log.Printf("Received Response for GetCredentialsInfo with datasetID: %s\n", responseCredential.GetDatasetId())
@@ -161,6 +147,7 @@ func GetCredentials(datasetID string) {
 		sCredential, _ := json.MarshalIndent(responseCredential, "", "\t")
 		log.Print(string(sCredential))
 		log.Println("***************************************************************")
+		return nil
 	}
 }
 
@@ -192,6 +179,21 @@ func main() {
 		datasetIDJson = "{\"catalog_id\":\"" + catalogID + "\",\"asset_id\":\"" + datasetID + "\"}"
 	}
 
-	GetMetadata(datasetIDJson)
-	GetCredentials(datasetIDJson)
+	err := GetMetadata(datasetIDJson)
+	if err != nil {
+		fmt.Printf("Error in GetCredentials:\n %v\n\n", err)
+		fmt.Printf("Error in GetCredentials Details:\n%+v\n\n", err)
+		// errors.Cause() provides access to original error.
+		fmt.Printf("Error in GetCredentials Cause: %v\n", errors.Cause(err))
+		fmt.Printf("Error in GetCredentials Extended Cause:\n%+v\n", errors.Cause(err))
+	}
+
+	err = GetCredentials(datasetIDJson)
+	if err != nil {
+		fmt.Printf("Error in GetCredentials: \n %v\n\n", err)
+		fmt.Printf("Error in GetCredentials Details: \n%+v\n\n", err)
+		// errors.Cause() provides access to original error.
+		fmt.Printf("Error in GetCredentials Cause: %v\n", errors.Cause(err))
+		fmt.Printf("Error in GetCredentials Details Extended Cause:\n%+v\n", errors.Cause(err))
+	}
 }

--- a/test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go
+++ b/test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go
@@ -15,7 +15,6 @@ import (
 	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
 
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
@@ -80,10 +79,10 @@ func GetMetadata(datasetID string) {
 		// you can do `int(status_code)` which will give you `3`
 		//
 		// Want to take specific action based on specific error?
-		if codes.InvalidArgument == errStatus.Code() {
-			// do your stuff here
-			log.Fatal()
-		}
+		// if codes.InvalidArgument == errStatus.Code() {
+		// 	do your stuff here
+		// 	log.Fatal()
+		// }
 	} else {
 		fmt.Println("***************************************************************")
 		log.Printf("Received Response for GetDatasetInfo with  datasetID: %s\n", r.GetDatasetId())
@@ -150,10 +149,10 @@ func GetCredentials(datasetID string) {
 		// you can do `int(status_code)` which will give you `3`
 		//
 		// Want to take specific action based on specific error?
-		if codes.InvalidArgument == errCredentialStatus.Code() {
-			// do your stuff here
-			log.Fatal()
-		}
+		// if codes.InvalidArgument == errCredentialStatus.Code() {
+		// 	do your stuff here
+		// 	log.Fatal()
+		// }
 	} else {
 		log.Println("***************************************************************")
 		log.Printf("Received Response for GetCredentialsInfo with datasetID: %s\n", responseCredential.GetDatasetId())
@@ -166,25 +165,25 @@ func GetCredentials(datasetID string) {
 }
 
 func main() {
-	//example 1: remote parquet
+	// example 1: remote parquet
 	// datasetID := "10a9fba1-b049-40d9-bac9-1a608c1e4774"
 	// catalogID := "591258ed-7461-47db-8eb6-1edf285c26cd"
 
-	//example 2: remote db2
+	// example 2: remote db2
 	// datasetID := "2d1b5352-1fbf-439b-8bb0-c1967ac484b9"
 	// catalogID := "1c080331-72da-4cea-8d06-5f075405cf17"
 
-	//example 3: remote csv,
-	//datasetID := "79aaff22-cfbe-470a-86b6-8f5125781a5c";
-	//catalogID := "1c080331-72da-4cea-8d06-5f075405cf17";
+	// example 3: remote csv,
+	// datasetID := "79aaff22-cfbe-470a-86b6-8f5125781a5c";
+	// catalogID := "1c080331-72da-4cea-8d06-5f075405cf17";
 
-	//example 4: local csv
+	// example 4: local csv
 	// datasetID := "cc17803b-163a-43db-97e3-323a8519c78f"
 	// dataSetID := "6c49313f-1207-4995-a957-5cd49c4e57ac"
 
-	//kafka
+	// kafka
 	catalogID := "87ffdca3-8b5d-4f77-99f9-0cb1fba1f73f"
-	datasetID := "01c6f0f0-9ffe-4ccc-ac07-409523755e72" //"466b5d7c-38c5-438c-8298-5c7e00e40638"
+	datasetID := "01c6f0f0-9ffe-4ccc-ac07-409523755e72" // "466b5d7c-38c5-438c-8298-5c7e00e40638"
 
 	var datasetIDJson string
 	if getEnv("CATALOG_PROVIDER_NAME") == "EGERIA" {

--- a/test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go
+++ b/test/services/pilot/catalog-credential-mock/mockup_pilot_catalog_con.go
@@ -76,16 +76,16 @@ func GetMetadata(datasetID string) error {
 		// lets print the error code which is `INVALID_ARGUMENT`
 		fmt.Println("Code:", errStatus.Code())
 		return errors.Wrap(err, "Error sending data to Catalog Connector")
-	} else {
-		fmt.Println("***************************************************************")
-		log.Printf("Received Response for GetDatasetInfo with  datasetID: %s\n", r.GetDatasetId())
-		fmt.Println("***************************************************************")
-		log.Printf("Response received from %s is given below:", catalogProviderName)
-		s, _ := json.MarshalIndent(r, "", "\t")
-		fmt.Print(string(s))
-		fmt.Println("***************************************************************")
-		return nil
 	}
+
+	fmt.Println("***************************************************************")
+	log.Printf("Received Response for GetDatasetInfo with  datasetID: %s\n", r.GetDatasetId())
+	fmt.Println("***************************************************************")
+	log.Printf("Response received from %s is given below:", catalogProviderName)
+	s, _ := json.MarshalIndent(r, "", "\t")
+	fmt.Print(string(s))
+	fmt.Println("***************************************************************")
+	return nil
 }
 
 func GetCredentials(datasetID string) error {
@@ -139,16 +139,16 @@ func GetCredentials(datasetID string) error {
 		// lets print the error code which is `INVALID_ARGUMENT`
 		log.Println("Code:", errCredentialStatus.Code())
 		return errors.Wrap(err, "Error sending data to Credentials Connector in GetCredentials")
-	} else {
-		log.Println("***************************************************************")
-		log.Printf("Received Response for GetCredentialsInfo with datasetID: %s\n", responseCredential.GetDatasetId())
-		log.Println("***************************************************************")
-		log.Printf("Response received from %s is given below:", credentialsProviderName)
-		sCredential, _ := json.MarshalIndent(responseCredential, "", "\t")
-		log.Print(string(sCredential))
-		log.Println("***************************************************************")
-		return nil
 	}
+
+	log.Println("***************************************************************")
+	log.Printf("Received Response for GetCredentialsInfo with datasetID: %s\n", responseCredential.GetDatasetId())
+	log.Println("***************************************************************")
+	log.Printf("Response received from %s is given below:", credentialsProviderName)
+	sCredential, _ := json.MarshalIndent(responseCredential, "", "\t")
+	log.Print(string(sCredential))
+	log.Println("***************************************************************")
+	return nil
 }
 
 func main() {

--- a/test/services/pilot/policy-manager-mock/mockup_pilot.go
+++ b/test/services/pilot/policy-manager-mock/mockup_pilot.go
@@ -80,16 +80,8 @@ func constructInputParameters() *pb.ApplicationContext {
 	// datasets = append(datasets, createDatasetTransferFirst(catalogIDDb2, datasetIDDb2))
 	// datasets = append(datasets, createDatasetTransferSecond(catalogIDDb2, datasetIDDb2))
 
-	// generalOperations := []*pb.AccessOperation{}
-
 	// applicationContext := &pb.ApplicationContext{PolicyManagerCredentials: policyManagerCredentials, AppInfo: applicationDetails, Datasets: datasets}
 	applicationContext := &pb.ApplicationContext{AppId: appID, AppInfo: applicationDetails, Datasets: datasets}
-
-	// ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
-	// ctx, cancel := context.WithTimeout(context.Background(), timeoutPeriod*time.Second)
-
-	// defer cancel()
-
 	log.Printf("Sending Application Context: ")
 	appContextStr, _ := json.MarshalIndent(applicationContext, "", "\t")
 	log.Print(string(appContextStr))
@@ -111,16 +103,13 @@ func main() {
 
 		// take specific action based on specific error?
 		if codes.InvalidArgument == errStatus.Code() {
-			// write customized code here
-			log.Fatal()
+			fmt.Println("InvalidArgument in mockup pilot")
+			return
 		}
 	} else {
 		log.Printf("Response received from Policy Compiler below:")
-
 		s, _ := json.MarshalIndent(r, "", "    ")
-
 		log.Print(string(s))
-
 		log.Println("2***************************************************************")
 	}
 
@@ -129,23 +118,19 @@ func main() {
 
 	if err != nil {
 		errStatus, _ := status.FromError(err)
-		fmt.Println("*********************************in error in  MockupPilot *****************************")
+		fmt.Println("*********************************in error in  MockupPilot for 2nd request *****************************")
 		fmt.Println("Message: ", errStatus.Message())
 		fmt.Println("Code: ", errStatus.Code())
 
 		// take specific action based on specific error?
 		if codes.InvalidArgument == errStatus.Code() {
-			// write customized code here
-			log.Fatal()
+			fmt.Println("InvalidArgument in mockup pilot for 2nd request")
+			return
 		}
 	} else {
-		log.Printf("Response received from Policy Compiler below:")
-
+		log.Printf("Response received from Policy Compiler below for 2nd request:")
 		s, _ := json.MarshalIndent(r, "", "    ")
-
 		log.Print(string(s))
-
-		log.Println("2***************************************************************")
 	}
 }
 

--- a/test/services/pilot/policy-manager-mock/mockup_pilot.go
+++ b/test/services/pilot/policy-manager-mock/mockup_pilot.go
@@ -35,28 +35,28 @@ func constructInputParameters() *pb.ApplicationContext {
 	fmt.Println("timeoutinseconds in MockupPilot: ", timeoutinseconds)
 
 	// Defining an applicationcontext
-	//ownerID := "999"
-	//credentialsStr := "{ ownerID: " + ownerID + "}"
+	// ownerID := "999"
+	// credentialsStr := "{ ownerID: " + ownerID + "}"
 
-	//example 0: local file
-	//catalogID := "cc17803b-163a-43db-97e3-323a8519c78f"  //democatalog
-	//datasetID := "6c49313f-1207-4995-a957-5cd49c4e57ac"
+	// example 0: local file
+	// catalogID := "cc17803b-163a-43db-97e3-323a8519c78f"  //democatalog
+	// datasetID := "6c49313f-1207-4995-a957-5cd49c4e57ac"
 
-	//example 1: remote parquet
+	// example 1: remote parquet
 	// datasetIDcos := "10a9fba1-b049-40d9-bac9-1a608c1e4774" //small.parq
 	// catalogIDcos := "591258ed-7461-47db-8eb6-1edf285c26cd" //EtyCatalog
 
-	//example 2: remote db2
+	// example 2: remote db2
 	// datasetIDDb2 := "2d1b5352-1fbf-439b-8bb0-c1967ac484b9" //Connection-Db2-NQD60833-SMALL
 	// catalogIDDb2 := "1c080331-72da-4cea-8d06-5f075405cf17" //catalog-suri
 
-	//example 3: remote csv,
+	// example 3: remote csv,
 	// datasetID := "79aaff22-cfbe-470a-86b6-8f5125781a5c"
 	// catalogID := "1c080331-72da-4cea-8d06-5f075405cf17"
 
-	//kafka
+	// kafka
 	catalogID := "87ffdca3-8b5d-4f77-99f9-0cb1fba1f73f"
-	datasetID := "01c6f0f0-9ffe-4ccc-ac07-409523755e72" //"988f7b32-2417-4b4f-b327-d4a63d110267" //"466b5d7c-38c5-438c-8298-5c7e00e40638"
+	datasetID := "01c6f0f0-9ffe-4ccc-ac07-409523755e72" // "988f7b32-2417-4b4f-b327-d4a63d110267" // "466b5d7c-38c5-438c-8298-5c7e00e40638"
 
 	var datasetIDJson string
 	if getEnv("CATALOG_PROVIDER_NAME") == "EGERIA" {
@@ -82,13 +82,13 @@ func constructInputParameters() *pb.ApplicationContext {
 
 	// generalOperations := []*pb.AccessOperation{}
 
-	//applicationContext := &pb.ApplicationContext{PolicyManagerCredentials: policyManagerCredentials, AppInfo: applicationDetails, Datasets: datasets}
+	// applicationContext := &pb.ApplicationContext{PolicyManagerCredentials: policyManagerCredentials, AppInfo: applicationDetails, Datasets: datasets}
 	applicationContext := &pb.ApplicationContext{AppId: appID, AppInfo: applicationDetails, Datasets: datasets}
 
-	//ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
-	//ctx, cancel := context.WithTimeout(context.Background(), timeoutPeriod*time.Second)
+	// ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	// ctx, cancel := context.WithTimeout(context.Background(), timeoutPeriod*time.Second)
 
-	//defer cancel()
+	// defer cancel()
 
 	log.Printf("Sending Application Context: ")
 	appContextStr, _ := json.MarshalIndent(applicationContext, "", "\t")
@@ -111,7 +111,7 @@ func main() {
 
 		// take specific action based on specific error?
 		if codes.InvalidArgument == errStatus.Code() {
-			//write customized code here
+			// write customized code here
 			log.Fatal()
 		}
 	} else {
@@ -135,7 +135,7 @@ func main() {
 
 		// take specific action based on specific error?
 		if codes.InvalidArgument == errStatus.Code() {
-			//write customized code here
+			// write customized code here
 			log.Fatal()
 		}
 	} else {

